Cross-Compile: macOS X1 (arm64)

ax3l opened the pull request:
* macOS: no arm64 ("X1") native runners yet

Note: this disable Fortran bindings due to a compile error.

<!--
Thank you for pull request.
Below are a few things we ask you kindly to self-check before getting a review. Remove checks that are not relevant.
-->
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [ ] Ensured the license file is being packaged.

<!--
Please note any issues this fixes using [closing keywords]( https://help.github.com/articles/closing-issues-using-keywords/ ):
-->

<!--
Please add any other relevant info below:
-->

ax3l committed:
Cross-Compile: macOS X1 (arm64)

* macOS: no arm64 ("X1") native runners yet

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.


I do have some suggestions for making it better though...


For **recipe**:

 * License is not an SPDX identifier (or a custom LicenseRef) nor an SPDX license expression.

Documentation on acceptable licenses can be found [here]( https://conda-forge.org/docs/maintainer/adding_pkgs.html#spdx-identifiers-and-expressions ).

ax3l commented:
@conda-forge-admin, please rerender

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe```) and found it was in an excellent condition.

ax3l commented:
@conda-forge-admin, please rerender

ax3l commented:
Hm...
If I build a c lib that needs python as a build script dependency but does not build python modules - is this still the right way to express this?
https://github.com/conda-forge/adios-feedstock/blob/8ed8311e0a1d41f08422cc8dfa0890a52d5a0b0b/recipe/conda_build_config.yaml#L10-L15
I updated locally to 3.8 instead of 3.6 but smithy still cannot render this
```
ValueError: variant config in /home/axel/src/conda-forge/adios-feedstock/recipe/conda_build_config.yaml is ambiguous because it
does not fully implement all zipped keys, or specifies a
subspace that is not fully implemented. To be clear:
.. we did not find ['3.8'] from {'mpi': ['nompi', 'mpich', 'openmpi'], 'pin_run_as_build': {'mpich': 'x.x', 'openmpi': 'x.x'}, 'python': ['3.8'], 'ignore_version': ['python']} in python:['3.6.* *_cpython', '3.7.* *_cpython', '3.8.* *_cpython']

```

ax3l committed:
Modernize build-only Python

We only need some python scripts executed during build.
(We don't want a full python bild matrix.)

ax3l commented:
@conda-forge-admin, please rerender

Commit:
MNT: Re-rendered with conda-build 3.21.3, conda-smithy 3.8.6, and conda-forge-pinning 2021.01.14.22.52.52

ax3l commented:
arm64 OSX compile issues with Fortran bindings:
```
$BUILD_PREFIX/bin/arm64-apple-darwin20.0.0-gfortran  -g -O2 -c -o core/adiosf_write_mod.o core/adiosf_write_mod.f90
core/adiosf_write_mod.f90:4109:98:
 4109 |         call adios_define_attribute_byvalue_f2c (group_id, attrname, path, adios_complex, nelems, values, err)
      |                                                                                                  1
......
 4122 |         call adios_define_attribute_byvalue_f2c (group_id, attrname, path, adios_double_complex, nelems, values, err)
      |                                                                                                         2
Error: Type mismatch between actual argument at (1) and actual argument at (2) (COMPLEX(4)/COMPLEX(8)).
core/adiosf_write_mod.f90:4096:97:

 4096 |         call adios_define_attribute_byvalue_f2c (group_id, attrname, path, adios_double, nelems, values, err)
      |                                                                                                 1
......
 4122 |         call adios_define_attribute_byvalue_f2c (group_id, attrname, path, adios_double_complex, nelems, values, err)
      |                                                                                                         2
Error: Type mismatch between actual argument at (1) and actual argument at (2) (REAL(8)/COMPLEX(8)).
core/adiosf_write_mod.f90:4083:95:

 4083 |         call adios_define_attribute_byvalue_f2c (group_id, attrname, path, adios_real, nelems, values, err)
      |                                                                                               1
......
 4122 |         call adios_define_attribute_byvalue_f2c (group_id, attrname, path, adios_double_complex, nelems, values, err)
      |                                                                                                         2
Error: Type mismatch between actual argument at (1) and actual argument at (2) (REAL(4)/COMPLEX(8)).
core/adiosf_write_mod.f90:4070:95:

 4070 |         call adios_define_attribute_byvalue_f2c (group_id, attrname, path, adios_long, nelems, values, err)
      |                                                                                               1
......
 4122 |         call adios_define_attribute_byvalue_f2c (group_id, attrname, path, adios_double_complex, nelems, values, err)
      |                                                                                                         2
Error: Type mismatch between actual argument at (1) and actual argument at (2) (INTEGER(8)/COMPLEX(8)).
core/adiosf_write_mod.f90:4057:98:

 4057 |         call adios_define_attribute_byvalue_f2c (group_id, attrname, path, adios_integer, nelems, values, err)
      |                                                                                                  1
......
 4122 |         call adios_define_attribute_byvalue_f2c (group_id, attrname, path, adios_double_complex, nelems, values, err)

[...]
```

ax3l committed:
Fortran Bindings: Disable for arm64 macOS

isuruf commented on recipe/build.sh:
```diff
@@ -26,12 +26,18 @@ else
 
+# Fortran bindings
+export ADIOS_FORTRAN="--enable-fortran"
+if [[ "$CONDA_BUILD_CROSS_COMPILATION" != "1" ]]; then
```
```suggestion
if [[ "$CONDA_BUILD_CROSS_COMPILATION" == "1" ]]; then
```

ax3l replied:
Ui, forgot to add.

I think I can add `-fallow-argument-mismatch` again to get this compiled...

ax3l committed:
gfortran (10+): -fallow-argument-mismatch

ax3l committed:
Check: Skip for cross-compile

ax3l commented:
Is the macOS arm64 OpenMPI hickup due to https://github.com/conda-forge/conda-forge-repodata-patches-feedstock/pull/110 ?

isuruf commented:
nope. see the changes I did in adios2 to allow cross compilation with openmpi.

ax3l commented:
Thanks, good point! https://github.com/conda-forge/adios2-feedstock/pull/21

ax3l committed:
Cross-Compile: OpenMPI in build

ax3l commented:
@conda-forge-admin, please rerender

isuruf committed:
Update build.sh
